app/controllers/application_controller.rb

```diff
@@ -1,6 +1,13 @@
 class ApplicationController < ActionController::API
   include ActionController::Serialization
+  include ActionController::HttpAuthentication::Basic::ControllerMethods
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   # protect_from_forgery with: :exception
+
+  def authenticated?
+    authenticate_or_request_with_http_basic {
+      |u,p| User.where( username: u, password: p ).present?
```

Comment:
Block argument expression is not on the same line as the block start.<br>Space missing after comma.<br>Space inside parentheses detected.